ainstein_radar_drivers/src/radar_interface_o79_can.cpp

```diff
@@ -67,56 +72,123 @@ namespace ainstein_radar_drivers
   unsigned int targets_received = 0;
   int target_type = -1;
   const ros::Duration t_raw_timeout = ros::Duration(1.0);
+  /* declare the Cartesian targets as global because each one spans two messages */
+  std::vector<ainstein_radar_drivers::RadarTargetCartesian> gtarget_cart;
   void RadarInterfaceO79CAN::dataMsgCallback( const can_msgs::Frame &msg )
   {
     if( msg.id == can_id_ )
       {
         // Parse out start of frame messages:
-        if( (msg.data[4]==0x00 || msg.data[4]==0x01) && msg.data[5]==0xFF && msg.data[6]==0xFF && msg.data[7]==0xFF )
+        if( (msg.data[4]==0x00 || msg.data[4]==0x01 || msg.data[4]==0x04) \
+            && msg.data[5]==0xFF && msg.data[6]==0xFF && msg.data[7]==0xFF )
           {
             ROS_DEBUG( "received start frame from radar" );
+            target_type = msg.data[4];
+            targets_received = 0;
+            targets_to_come = (msg.data[2] << 8) | msg.data[3];
+
             // clear radar data message arrays here
             radar_data_msg_ptr_raw_->header.stamp = ros::Time::now();
             radar_data_msg_ptr_tracked_->header.stamp = ros::Time::now();
 
-            radar_data_msg_ptr_raw_->targets.clear();
-            radar_data_msg_ptr_tracked_->targets.clear();
-            targets_to_come = (msg.data[2] << 8) | msg.data[3];
-            target_type = msg.data[4];
-            targets_received = 0;
+            if( target_type == 0 || target_type == 1 )
```

Comment:
We should really use constants (defines or similar) instead of these magic IDs for things like target_type, but I'm guilty of that all over the place too. Maybe something I can add to my PR.